packages/fxa-email-service/src/db/delivery_problems/test.rs

```diff
@@ -132,14 +132,19 @@ fn check_soft_bounce() {
 pub struct DbMockBounceSoft;
 
 impl Db for DbMockBounceSoft {
-    fn get_bounces(&self, _address: &EmailAddress) -> AppResult<Vec<LegacyDeliveryProblem>> {
+    fn get_bounces(&self, address: &EmailAddress) -> AppResult<Vec<LegacyDeliveryProblem>> {
         let now = now_as_milliseconds();
-        Ok(vec![LegacyDeliveryProblem {
-            address: "foo@example.com".parse().unwrap(),
-            problem_type: ProblemType::SoftBounce,
-            problem_subtype: ProblemSubtype::Undetermined,
-            created_at: now - DAY + SECOND * 2,
-        }])
+        let result = if address.as_ref() == "foo@example.com" {
```

Comment:
I'm leaning on these mocks below, so tweaked them to be more selective about the errors they return.